{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's start lookinf for classic citations at the EuropePMC database.\n",
    "\n",
    "Once more, we will use naive definitions of citation classics. \n",
    "\n",
    "Classics are all cited more than 150 times. \n",
    "\n",
    "Big classics: more than 500 citations\n",
    "\n",
    "Medium classics: at least 300 citations, less than 500\n",
    "\n",
    "Small classics: at least 150 citations, less than 300\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "from pandas.io.json import json_normalize\n",
    "\n",
    "\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "pmc_query = \"https://www.ebi.ac.uk/europepmc/webservices/rest/search?query=%22single-cell%20RNA-seq%22&resultType=lite&cursorMark=*&pageSize=100&sort=CITED%20desc&format=json\"\n",
    "page = requests.get(pmc_query)\n",
    "\n",
    "data = json.loads(page.text)\n",
    "df = json_normalize(data['resultList']['result'])\n",
    "\n",
    "\n",
    "single_cell_papers = df[['pmid', 'title', 'authorString', 'journalTitle', 'citedByCount', 'pubYear']]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3/dist-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>title</th>\n",
       "      <th>authorString</th>\n",
       "      <th>journalTitle</th>\n",
       "      <th>citedByCount</th>\n",
       "      <th>pubYear</th>\n",
       "      <th>firstAuthor</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26000488</td>\n",
       "      <td>Highly Parallel Genome-wide Expression Profili...</td>\n",
       "      <td>Macosko EZ, Basu A, Satija R, Nemesh J, Shekha...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>1092</td>\n",
       "      <td>2015</td>\n",
       "      <td>Macosko EZ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>24925914</td>\n",
       "      <td>Single-cell RNA-seq highlights intratumoral he...</td>\n",
       "      <td>Patel AP, Tirosh I, Trombetta JJ, Shalek AK, G...</td>\n",
       "      <td>Science</td>\n",
       "      <td>938</td>\n",
       "      <td>2014</td>\n",
       "      <td>Patel AP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>25700174</td>\n",
       "      <td>Brain structure. Cell types in the mouse corte...</td>\n",
       "      <td>Zeisel A, Muñoz-Manchado AB, Codeluppi S, Lönn...</td>\n",
       "      <td>Science</td>\n",
       "      <td>726</td>\n",
       "      <td>2015</td>\n",
       "      <td>Zeisel A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26000487</td>\n",
       "      <td>Droplet barcoding for single-cell transcriptom...</td>\n",
       "      <td>Klein AM, Mazutis L, Akartuna I, Tallapragada ...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>582</td>\n",
       "      <td>2015</td>\n",
       "      <td>Klein AM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>24658644</td>\n",
       "      <td>The dynamics and regulators of cell fate decis...</td>\n",
       "      <td>Trapnell C, Cacchiarelli D, Grimsby J, Pokhare...</td>\n",
       "      <td>Nat Biotechnol</td>\n",
       "      <td>569</td>\n",
       "      <td>2014</td>\n",
       "      <td>Trapnell C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>25480296</td>\n",
       "      <td>Tissue-resident macrophage enhancer landscapes...</td>\n",
       "      <td>Lavin Y, Winter D, Blecher-Gonen R, David E, K...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>545</td>\n",
       "      <td>2014</td>\n",
       "      <td>Lavin Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>24385147</td>\n",
       "      <td>Full-length RNA-seq from single cells using Sm...</td>\n",
       "      <td>Picelli S, Faridani OR, Björklund AK, Winberg ...</td>\n",
       "      <td>Nat Protoc</td>\n",
       "      <td>519</td>\n",
       "      <td>2014</td>\n",
       "      <td>Picelli S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>24531970</td>\n",
       "      <td>Massively parallel single-cell RNA-seq for mar...</td>\n",
       "      <td>Jaitin DA, Kenigsberg E, Keren-Shaul H, Elefan...</td>\n",
       "      <td>Science</td>\n",
       "      <td>500</td>\n",
       "      <td>2014</td>\n",
       "      <td>Jaitin DA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>22820318</td>\n",
       "      <td>Full-length mRNA-Seq from single-cell levels o...</td>\n",
       "      <td>Ramsköld D, Luo S, Wang YC, Li R, Deng Q, Fari...</td>\n",
       "      <td>Nat Biotechnol</td>\n",
       "      <td>483</td>\n",
       "      <td>2012</td>\n",
       "      <td>Ramsköld D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>23685454</td>\n",
       "      <td>Single-cell transcriptomics reveals bimodality...</td>\n",
       "      <td>Shalek AK, Satija R, Adiconis X, Gertner RS, G...</td>\n",
       "      <td>Nature</td>\n",
       "      <td>477</td>\n",
       "      <td>2013</td>\n",
       "      <td>Shalek AK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>24739965</td>\n",
       "      <td>Reconstructing lineage hierarchies of the dist...</td>\n",
       "      <td>Treutlein B, Brownfield DG, Wu AR, Neff NF, Ma...</td>\n",
       "      <td>Nature</td>\n",
       "      <td>473</td>\n",
       "      <td>2014</td>\n",
       "      <td>Treutlein B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>27124452</td>\n",
       "      <td>Dissecting the multicellular ecosystem of meta...</td>\n",
       "      <td>Tirosh I, Izar B, Prakadan SM, Wadsworth MH, T...</td>\n",
       "      <td>Science</td>\n",
       "      <td>465</td>\n",
       "      <td>2016</td>\n",
       "      <td>Tirosh I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>23934149</td>\n",
       "      <td>Single-cell RNA-Seq profiling of human preimpl...</td>\n",
       "      <td>Yan L, Yang M, Guo H, Yang L, Wu J, Li R, Liu ...</td>\n",
       "      <td>Nat Struct Mol Biol</td>\n",
       "      <td>451</td>\n",
       "      <td>2013</td>\n",
       "      <td>Yan L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>25420068</td>\n",
       "      <td>Unbiased classification of sensory neuron type...</td>\n",
       "      <td>Usoskin D, Furlan A, Islam S, Abdo H, Lönnerbe...</td>\n",
       "      <td>Nat Neurosci</td>\n",
       "      <td>417</td>\n",
       "      <td>2015</td>\n",
       "      <td>Usoskin D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>25867923</td>\n",
       "      <td>Spatial reconstruction of single-cell gene exp...</td>\n",
       "      <td>Satija R, Farrell JA, Gennert D, Schier AF, Re...</td>\n",
       "      <td>Nat Biotechnol</td>\n",
       "      <td>399</td>\n",
       "      <td>2015</td>\n",
       "      <td>Satija R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>26813401</td>\n",
       "      <td>A survey of best practices for RNA-seq data an...</td>\n",
       "      <td>Conesa A, Madrigal P, Tarazona S, Gomez-Cabrer...</td>\n",
       "      <td>Genome Biol</td>\n",
       "      <td>387</td>\n",
       "      <td>2016</td>\n",
       "      <td>Conesa A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>24439369</td>\n",
       "      <td>Reversing DNA methylation: mechanisms, genomic...</td>\n",
       "      <td>Wu H, Zhang Y.</td>\n",
       "      <td>Cell</td>\n",
       "      <td>378</td>\n",
       "      <td>2014</td>\n",
       "      <td>Wu H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>22939981</td>\n",
       "      <td>CEL-Seq: single-cell RNA-Seq by multiplexed li...</td>\n",
       "      <td>Hashimshony T, Wagner F, Sher N, Yanai I.</td>\n",
       "      <td>Cell Rep</td>\n",
       "      <td>365</td>\n",
       "      <td>2012</td>\n",
       "      <td>Hashimshony T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>24919153</td>\n",
       "      <td>Single-cell RNA-seq reveals dynamic paracrine ...</td>\n",
       "      <td>Shalek AK, Satija R, Shuga J, Trombetta JJ, Ge...</td>\n",
       "      <td>Nature</td>\n",
       "      <td>346</td>\n",
       "      <td>2014</td>\n",
       "      <td>Shalek AK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>28091601</td>\n",
       "      <td>Massively parallel digital transcriptional pro...</td>\n",
       "      <td>Zheng GX, Terry JM, Belgrader P, Ryvkin P, Ben...</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>340</td>\n",
       "      <td>2017</td>\n",
       "      <td>Zheng GX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>24408435</td>\n",
       "      <td>Single-cell RNA-seq reveals dynamic, random mo...</td>\n",
       "      <td>Deng Q, Ramsköld D, Reinius B, Sandberg R.</td>\n",
       "      <td>Science</td>\n",
       "      <td>335</td>\n",
       "      <td>2014</td>\n",
       "      <td>Deng Q</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>24363023</td>\n",
       "      <td>Quantitative single-cell RNA-seq with unique m...</td>\n",
       "      <td>Islam S, Zeisel A, Joost S, La Manno G, Zajac ...</td>\n",
       "      <td>Nat Methods</td>\n",
       "      <td>327</td>\n",
       "      <td>2014</td>\n",
       "      <td>Islam S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>28602351</td>\n",
       "      <td>A Unique Microglia Type Associated with Restri...</td>\n",
       "      <td>Keren-Shaul H, Spinrad A, Weiner A, Matcovitch...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>322</td>\n",
       "      <td>2017</td>\n",
       "      <td>Keren-Shaul H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>23223451</td>\n",
       "      <td>Germline DNA demethylation dynamics and imprin...</td>\n",
       "      <td>Hackett JA, Sengupta R, Zylicz JJ, Murakami K,...</td>\n",
       "      <td>Science</td>\n",
       "      <td>322</td>\n",
       "      <td>2013</td>\n",
       "      <td>Hackett JA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>23892778</td>\n",
       "      <td>Genetic programs in human and mouse early embr...</td>\n",
       "      <td>Xue Z, Huang K, Cai C, Cai L, Jiang CY, Feng Y...</td>\n",
       "      <td>Nature</td>\n",
       "      <td>321</td>\n",
       "      <td>2013</td>\n",
       "      <td>Xue Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>21543516</td>\n",
       "      <td>Characterization of the single-cell transcript...</td>\n",
       "      <td>Islam S, Kjällquist U, Moliner A, Zajac P, Fan...</td>\n",
       "      <td>Genome Res</td>\n",
       "      <td>321</td>\n",
       "      <td>2011</td>\n",
       "      <td>Islam S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>26727548</td>\n",
       "      <td>Adult mouse cortical cell taxonomy revealed by...</td>\n",
       "      <td>Tasic B, Menon V, Nguyen TN, Kim TK, Jarsky T,...</td>\n",
       "      <td>Nat Neurosci</td>\n",
       "      <td>308</td>\n",
       "      <td>2016</td>\n",
       "      <td>Tasic B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>26083756</td>\n",
       "      <td>Single-cell chromatin accessibility reveals pr...</td>\n",
       "      <td>Buenrostro JD, Wu B, Litzenburger UM, Ruff D, ...</td>\n",
       "      <td>Nature</td>\n",
       "      <td>305</td>\n",
       "      <td>2015</td>\n",
       "      <td>Buenrostro JD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>25215486</td>\n",
       "      <td>Resetting transcription factor control circuit...</td>\n",
       "      <td>Takashima Y, Guo G, Loos R, Nichols J, Ficz G,...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>300</td>\n",
       "      <td>2014</td>\n",
       "      <td>Takashima Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>24056876</td>\n",
       "      <td>Accounting for technical noise in single-cell ...</td>\n",
       "      <td>Brennecke P, Anders S, Kim JK, Kołodziejczyk A...</td>\n",
       "      <td>Nat Methods</td>\n",
       "      <td>294</td>\n",
       "      <td>2013</td>\n",
       "      <td>Brennecke P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>25628217</td>\n",
       "      <td>Computational and analytical challenges in sin...</td>\n",
       "      <td>Stegle O, Teichmann SA, Marioni JC.</td>\n",
       "      <td>Nat Rev Genet</td>\n",
       "      <td>273</td>\n",
       "      <td>2015</td>\n",
       "      <td>Stegle O</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>24836921</td>\n",
       "      <td>Bayesian approach to single-cell differential ...</td>\n",
       "      <td>Kharchenko PV, Silberstein L, Scadden DT.</td>\n",
       "      <td>Nat Methods</td>\n",
       "      <td>269</td>\n",
       "      <td>2014</td>\n",
       "      <td>Kharchenko PV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>20452321</td>\n",
       "      <td>Tracing the derivation of embryonic stem cells...</td>\n",
       "      <td>Tang F, Barbacioru C, Bao S, Lee C, Nordman E,...</td>\n",
       "      <td>Cell Stem Cell</td>\n",
       "      <td>266</td>\n",
       "      <td>2010</td>\n",
       "      <td>Tang F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>26721685</td>\n",
       "      <td>Biogenesis and function of tRNA fragments duri...</td>\n",
       "      <td>Sharma U, Conine CC, Shea JM, Boskovic A, Derr...</td>\n",
       "      <td>Science</td>\n",
       "      <td>251</td>\n",
       "      <td>2016</td>\n",
       "      <td>Sharma U</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>20203668</td>\n",
       "      <td>RNA-Seq analysis to capture the transcriptome ...</td>\n",
       "      <td>Tang F, Barbacioru C, Nordman E, Li B, Xu N, B...</td>\n",
       "      <td>Nat Protoc</td>\n",
       "      <td>233</td>\n",
       "      <td>2010</td>\n",
       "      <td>Tang F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>28428369</td>\n",
       "      <td>Single-cell RNA-seq reveals new types of human...</td>\n",
       "      <td>Villani AC, Satija R, Reynolds G, Sarkizova S,...</td>\n",
       "      <td>Science</td>\n",
       "      <td>231</td>\n",
       "      <td>2017</td>\n",
       "      <td>Villani AC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>26299571</td>\n",
       "      <td>Single-Cell RNA-Seq with Waterfall Reveals Mol...</td>\n",
       "      <td>Shin J, Berg DA, Zhu Y, Shin JY, Song J, Bonag...</td>\n",
       "      <td>Cell Stem Cell</td>\n",
       "      <td>217</td>\n",
       "      <td>2015</td>\n",
       "      <td>Shin J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>25105578</td>\n",
       "      <td>Repair and regeneration of the respiratory sys...</td>\n",
       "      <td>Hogan BL, Barkauskas CE, Chapman HA, Epstein J...</td>\n",
       "      <td>Cell Stem Cell</td>\n",
       "      <td>212</td>\n",
       "      <td>2014</td>\n",
       "      <td>Hogan BL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>25953818</td>\n",
       "      <td>Multiplex single cell profiling of chromatin a...</td>\n",
       "      <td>Cusanovich DA, Daza R, Adey A, Pliner HA, Chri...</td>\n",
       "      <td>Science</td>\n",
       "      <td>201</td>\n",
       "      <td>2015</td>\n",
       "      <td>Cusanovich DA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>26481351</td>\n",
       "      <td>Expression Atlas update--an integrated databas...</td>\n",
       "      <td>Petryszak R, Keays M, Tang YA, Fonseca NA, Bar...</td>\n",
       "      <td>Nucleic Acids Res</td>\n",
       "      <td>199</td>\n",
       "      <td>2016</td>\n",
       "      <td>Petryszak R</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>26593421</td>\n",
       "      <td>RNA mis-splicing in disease.</td>\n",
       "      <td>Scotti MM, Swanson MS.</td>\n",
       "      <td>Nat Rev Genet</td>\n",
       "      <td>197</td>\n",
       "      <td>2016</td>\n",
       "      <td>Scotti MM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>26458175</td>\n",
       "      <td>Single-cell ChIP-seq reveals cell subpopulatio...</td>\n",
       "      <td>Rotem A, Ram O, Shoresh N, Sperling RA, Goren ...</td>\n",
       "      <td>Nat Biotechnol</td>\n",
       "      <td>197</td>\n",
       "      <td>2015</td>\n",
       "      <td>Rotem A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>27038591</td>\n",
       "      <td>Expression Analysis Highlights AXL as a Candid...</td>\n",
       "      <td>Nowakowski TJ, Pollen AA, Di Lullo E, Sandoval...</td>\n",
       "      <td>Cell Stem Cell</td>\n",
       "      <td>189</td>\n",
       "      <td>2016</td>\n",
       "      <td>Nowakowski TJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>25053837</td>\n",
       "      <td>Single-cell RNA-seq: advances and future chall...</td>\n",
       "      <td>Saliba AE, Westermann AJ, Gorski SA, Vogel J.</td>\n",
       "      <td>Nucleic Acids Res</td>\n",
       "      <td>185</td>\n",
       "      <td>2014</td>\n",
       "      <td>Saliba AE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>24299736</td>\n",
       "      <td>From single-cell to cell-pool transcriptomes: ...</td>\n",
       "      <td>Marinov GK, Williams BA, McCue K, Schroth GP, ...</td>\n",
       "      <td>Genome Res</td>\n",
       "      <td>183</td>\n",
       "      <td>2014</td>\n",
       "      <td>Marinov GK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>27667667</td>\n",
       "      <td>Single-Cell Transcriptome Profiling of Human P...</td>\n",
       "      <td>Segerstolpe Å, Palasantza A, Eliasson P, Ander...</td>\n",
       "      <td>Cell Metab</td>\n",
       "      <td>177</td>\n",
       "      <td>2016</td>\n",
       "      <td>Segerstolpe Å</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>26248267</td>\n",
       "      <td>Toward understanding and exploiting tumor hete...</td>\n",
       "      <td>Alizadeh AA, Aranda V, Bardelli A, Blanpain C,...</td>\n",
       "      <td>Nat Med</td>\n",
       "      <td>176</td>\n",
       "      <td>2015</td>\n",
       "      <td>Alizadeh AA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>29790989</td>\n",
       "      <td>The Galaxy platform for accessible, reproducib...</td>\n",
       "      <td>Afgan E, Baker D, Batut B, van den Beek M, Bou...</td>\n",
       "      <td>Nucleic Acids Res</td>\n",
       "      <td>175</td>\n",
       "      <td>2018</td>\n",
       "      <td>Afgan E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>27565351</td>\n",
       "      <td>Comprehensive Classification of Retinal Bipola...</td>\n",
       "      <td>Shekhar K, Lapan SW, Whitney IE, Tran NM, Maco...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>170</td>\n",
       "      <td>2016</td>\n",
       "      <td>Shekhar K</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>27135602</td>\n",
       "      <td>Origin, fate and dynamics of macrophages at ce...</td>\n",
       "      <td>Goldmann T, Wieghofer P, Jordão MJ, Prutek F, ...</td>\n",
       "      <td>Nat Immunol</td>\n",
       "      <td>170</td>\n",
       "      <td>2016</td>\n",
       "      <td>Goldmann T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>26406371</td>\n",
       "      <td>Molecular identity of human outer radial glia ...</td>\n",
       "      <td>Pollen AA, Nowakowski TJ, Chen J, Retallack H,...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>163</td>\n",
       "      <td>2015</td>\n",
       "      <td>Pollen AA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>26644564</td>\n",
       "      <td>Human cerebral organoids recapitulate gene exp...</td>\n",
       "      <td>Camp JG, Badsha F, Florio M, Kanton S, Gerber ...</td>\n",
       "      <td>Proc Natl Acad Sci U S A</td>\n",
       "      <td>161</td>\n",
       "      <td>2015</td>\n",
       "      <td>Camp JG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>26653891</td>\n",
       "      <td>MAST: a flexible statistical framework for ass...</td>\n",
       "      <td>Finak G, McDavid A, Yajima M, Deng J, Gersuk V...</td>\n",
       "      <td>Genome Biol</td>\n",
       "      <td>160</td>\n",
       "      <td>2015</td>\n",
       "      <td>Finak G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>25533958</td>\n",
       "      <td>Lineage-negative progenitors mobilize to regen...</td>\n",
       "      <td>Vaughan AE, Brumwell AN, Xi Y, Gotts JE, Brown...</td>\n",
       "      <td>Nature</td>\n",
       "      <td>160</td>\n",
       "      <td>2015</td>\n",
       "      <td>Vaughan AE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>28346451</td>\n",
       "      <td>SC3: consensus clustering of single-cell RNA-s...</td>\n",
       "      <td>Kiselev VY, Kirschner K, Schaub MT, Andrews T,...</td>\n",
       "      <td>Nat Methods</td>\n",
       "      <td>158</td>\n",
       "      <td>2017</td>\n",
       "      <td>Kiselev VY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>27984732</td>\n",
       "      <td>Perturb-Seq: Dissecting Molecular Circuits wit...</td>\n",
       "      <td>Dixit A, Parnas O, Li B, Chen J, Fulco CP, Jer...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>158</td>\n",
       "      <td>2016</td>\n",
       "      <td>Dixit A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>28212749</td>\n",
       "      <td>Comparative Analysis of Single-Cell RNA Sequen...</td>\n",
       "      <td>Ziegenhain C, Vieth B, Parekh S, Reinius B, Gu...</td>\n",
       "      <td>Mol Cell</td>\n",
       "      <td>157</td>\n",
       "      <td>2017</td>\n",
       "      <td>Ziegenhain C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>25471879</td>\n",
       "      <td>Deconstructing transcriptional heterogeneity i...</td>\n",
       "      <td>Kumar RM, Cahan P, Shalek AK, Satija R, DaleyK...</td>\n",
       "      <td>Nature</td>\n",
       "      <td>155</td>\n",
       "      <td>2014</td>\n",
       "      <td>Kumar RM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>26431181</td>\n",
       "      <td>Adult Mammalian Neural Stem Cells and Neurogen...</td>\n",
       "      <td>Bond AM, Ming GL, Song H.</td>\n",
       "      <td>Cell Stem Cell</td>\n",
       "      <td>151</td>\n",
       "      <td>2015</td>\n",
       "      <td>Bond AM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>24497842</td>\n",
       "      <td>Single cell genomics: advances and future pers...</td>\n",
       "      <td>Macaulay IC, Voet T.</td>\n",
       "      <td>PLoS Genet</td>\n",
       "      <td>151</td>\n",
       "      <td>2014</td>\n",
       "      <td>Macaulay IC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pmid                                              title  \\\n",
       "0   26000488  Highly Parallel Genome-wide Expression Profili...   \n",
       "1   24925914  Single-cell RNA-seq highlights intratumoral he...   \n",
       "2   25700174  Brain structure. Cell types in the mouse corte...   \n",
       "3   26000487  Droplet barcoding for single-cell transcriptom...   \n",
       "4   24658644  The dynamics and regulators of cell fate decis...   \n",
       "5   25480296  Tissue-resident macrophage enhancer landscapes...   \n",
       "6   24385147  Full-length RNA-seq from single cells using Sm...   \n",
       "7   24531970  Massively parallel single-cell RNA-seq for mar...   \n",
       "8   22820318  Full-length mRNA-Seq from single-cell levels o...   \n",
       "9   23685454  Single-cell transcriptomics reveals bimodality...   \n",
       "10  24739965  Reconstructing lineage hierarchies of the dist...   \n",
       "11  27124452  Dissecting the multicellular ecosystem of meta...   \n",
       "12  23934149  Single-cell RNA-Seq profiling of human preimpl...   \n",
       "13  25420068  Unbiased classification of sensory neuron type...   \n",
       "14  25867923  Spatial reconstruction of single-cell gene exp...   \n",
       "15  26813401  A survey of best practices for RNA-seq data an...   \n",
       "16  24439369  Reversing DNA methylation: mechanisms, genomic...   \n",
       "17  22939981  CEL-Seq: single-cell RNA-Seq by multiplexed li...   \n",
       "18  24919153  Single-cell RNA-seq reveals dynamic paracrine ...   \n",
       "19  28091601  Massively parallel digital transcriptional pro...   \n",
       "20  24408435  Single-cell RNA-seq reveals dynamic, random mo...   \n",
       "21  24363023  Quantitative single-cell RNA-seq with unique m...   \n",
       "22  28602351  A Unique Microglia Type Associated with Restri...   \n",
       "23  23223451  Germline DNA demethylation dynamics and imprin...   \n",
       "24  23892778  Genetic programs in human and mouse early embr...   \n",
       "25  21543516  Characterization of the single-cell transcript...   \n",
       "26  26727548  Adult mouse cortical cell taxonomy revealed by...   \n",
       "27  26083756  Single-cell chromatin accessibility reveals pr...   \n",
       "28  25215486  Resetting transcription factor control circuit...   \n",
       "29  24056876  Accounting for technical noise in single-cell ...   \n",
       "..       ...                                                ...   \n",
       "34  25628217  Computational and analytical challenges in sin...   \n",
       "35  24836921  Bayesian approach to single-cell differential ...   \n",
       "36  20452321  Tracing the derivation of embryonic stem cells...   \n",
       "37  26721685  Biogenesis and function of tRNA fragments duri...   \n",
       "38  20203668  RNA-Seq analysis to capture the transcriptome ...   \n",
       "39  28428369  Single-cell RNA-seq reveals new types of human...   \n",
       "40  26299571  Single-Cell RNA-Seq with Waterfall Reveals Mol...   \n",
       "41  25105578  Repair and regeneration of the respiratory sys...   \n",
       "42  25953818  Multiplex single cell profiling of chromatin a...   \n",
       "43  26481351  Expression Atlas update--an integrated databas...   \n",
       "44  26593421                       RNA mis-splicing in disease.   \n",
       "45  26458175  Single-cell ChIP-seq reveals cell subpopulatio...   \n",
       "46  27038591  Expression Analysis Highlights AXL as a Candid...   \n",
       "47  25053837  Single-cell RNA-seq: advances and future chall...   \n",
       "48  24299736  From single-cell to cell-pool transcriptomes: ...   \n",
       "49  27667667  Single-Cell Transcriptome Profiling of Human P...   \n",
       "50  26248267  Toward understanding and exploiting tumor hete...   \n",
       "51  29790989  The Galaxy platform for accessible, reproducib...   \n",
       "52  27565351  Comprehensive Classification of Retinal Bipola...   \n",
       "53  27135602  Origin, fate and dynamics of macrophages at ce...   \n",
       "54  26406371  Molecular identity of human outer radial glia ...   \n",
       "55  26644564  Human cerebral organoids recapitulate gene exp...   \n",
       "56  26653891  MAST: a flexible statistical framework for ass...   \n",
       "57  25533958  Lineage-negative progenitors mobilize to regen...   \n",
       "58  28346451  SC3: consensus clustering of single-cell RNA-s...   \n",
       "59  27984732  Perturb-Seq: Dissecting Molecular Circuits wit...   \n",
       "60  28212749  Comparative Analysis of Single-Cell RNA Sequen...   \n",
       "61  25471879  Deconstructing transcriptional heterogeneity i...   \n",
       "62  26431181  Adult Mammalian Neural Stem Cells and Neurogen...   \n",
       "63  24497842  Single cell genomics: advances and future pers...   \n",
       "\n",
       "                                         authorString  \\\n",
       "0   Macosko EZ, Basu A, Satija R, Nemesh J, Shekha...   \n",
       "1   Patel AP, Tirosh I, Trombetta JJ, Shalek AK, G...   \n",
       "2   Zeisel A, Muñoz-Manchado AB, Codeluppi S, Lönn...   \n",
       "3   Klein AM, Mazutis L, Akartuna I, Tallapragada ...   \n",
       "4   Trapnell C, Cacchiarelli D, Grimsby J, Pokhare...   \n",
       "5   Lavin Y, Winter D, Blecher-Gonen R, David E, K...   \n",
       "6   Picelli S, Faridani OR, Björklund AK, Winberg ...   \n",
       "7   Jaitin DA, Kenigsberg E, Keren-Shaul H, Elefan...   \n",
       "8   Ramsköld D, Luo S, Wang YC, Li R, Deng Q, Fari...   \n",
       "9   Shalek AK, Satija R, Adiconis X, Gertner RS, G...   \n",
       "10  Treutlein B, Brownfield DG, Wu AR, Neff NF, Ma...   \n",
       "11  Tirosh I, Izar B, Prakadan SM, Wadsworth MH, T...   \n",
       "12  Yan L, Yang M, Guo H, Yang L, Wu J, Li R, Liu ...   \n",
       "13  Usoskin D, Furlan A, Islam S, Abdo H, Lönnerbe...   \n",
       "14  Satija R, Farrell JA, Gennert D, Schier AF, Re...   \n",
       "15  Conesa A, Madrigal P, Tarazona S, Gomez-Cabrer...   \n",
       "16                                     Wu H, Zhang Y.   \n",
       "17          Hashimshony T, Wagner F, Sher N, Yanai I.   \n",
       "18  Shalek AK, Satija R, Shuga J, Trombetta JJ, Ge...   \n",
       "19  Zheng GX, Terry JM, Belgrader P, Ryvkin P, Ben...   \n",
       "20         Deng Q, Ramsköld D, Reinius B, Sandberg R.   \n",
       "21  Islam S, Zeisel A, Joost S, La Manno G, Zajac ...   \n",
       "22  Keren-Shaul H, Spinrad A, Weiner A, Matcovitch...   \n",
       "23  Hackett JA, Sengupta R, Zylicz JJ, Murakami K,...   \n",
       "24  Xue Z, Huang K, Cai C, Cai L, Jiang CY, Feng Y...   \n",
       "25  Islam S, Kjällquist U, Moliner A, Zajac P, Fan...   \n",
       "26  Tasic B, Menon V, Nguyen TN, Kim TK, Jarsky T,...   \n",
       "27  Buenrostro JD, Wu B, Litzenburger UM, Ruff D, ...   \n",
       "28  Takashima Y, Guo G, Loos R, Nichols J, Ficz G,...   \n",
       "29  Brennecke P, Anders S, Kim JK, Kołodziejczyk A...   \n",
       "..                                                ...   \n",
       "34                Stegle O, Teichmann SA, Marioni JC.   \n",
       "35          Kharchenko PV, Silberstein L, Scadden DT.   \n",
       "36  Tang F, Barbacioru C, Bao S, Lee C, Nordman E,...   \n",
       "37  Sharma U, Conine CC, Shea JM, Boskovic A, Derr...   \n",
       "38  Tang F, Barbacioru C, Nordman E, Li B, Xu N, B...   \n",
       "39  Villani AC, Satija R, Reynolds G, Sarkizova S,...   \n",
       "40  Shin J, Berg DA, Zhu Y, Shin JY, Song J, Bonag...   \n",
       "41  Hogan BL, Barkauskas CE, Chapman HA, Epstein J...   \n",
       "42  Cusanovich DA, Daza R, Adey A, Pliner HA, Chri...   \n",
       "43  Petryszak R, Keays M, Tang YA, Fonseca NA, Bar...   \n",
       "44                             Scotti MM, Swanson MS.   \n",
       "45  Rotem A, Ram O, Shoresh N, Sperling RA, Goren ...   \n",
       "46  Nowakowski TJ, Pollen AA, Di Lullo E, Sandoval...   \n",
       "47      Saliba AE, Westermann AJ, Gorski SA, Vogel J.   \n",
       "48  Marinov GK, Williams BA, McCue K, Schroth GP, ...   \n",
       "49  Segerstolpe Å, Palasantza A, Eliasson P, Ander...   \n",
       "50  Alizadeh AA, Aranda V, Bardelli A, Blanpain C,...   \n",
       "51  Afgan E, Baker D, Batut B, van den Beek M, Bou...   \n",
       "52  Shekhar K, Lapan SW, Whitney IE, Tran NM, Maco...   \n",
       "53  Goldmann T, Wieghofer P, Jordão MJ, Prutek F, ...   \n",
       "54  Pollen AA, Nowakowski TJ, Chen J, Retallack H,...   \n",
       "55  Camp JG, Badsha F, Florio M, Kanton S, Gerber ...   \n",
       "56  Finak G, McDavid A, Yajima M, Deng J, Gersuk V...   \n",
       "57  Vaughan AE, Brumwell AN, Xi Y, Gotts JE, Brown...   \n",
       "58  Kiselev VY, Kirschner K, Schaub MT, Andrews T,...   \n",
       "59  Dixit A, Parnas O, Li B, Chen J, Fulco CP, Jer...   \n",
       "60  Ziegenhain C, Vieth B, Parekh S, Reinius B, Gu...   \n",
       "61  Kumar RM, Cahan P, Shalek AK, Satija R, DaleyK...   \n",
       "62                          Bond AM, Ming GL, Song H.   \n",
       "63                               Macaulay IC, Voet T.   \n",
       "\n",
       "                journalTitle  citedByCount pubYear    firstAuthor  \n",
       "0                       Cell          1092    2015     Macosko EZ  \n",
       "1                    Science           938    2014       Patel AP  \n",
       "2                    Science           726    2015       Zeisel A  \n",
       "3                       Cell           582    2015       Klein AM  \n",
       "4             Nat Biotechnol           569    2014     Trapnell C  \n",
       "5                       Cell           545    2014        Lavin Y  \n",
       "6                 Nat Protoc           519    2014      Picelli S  \n",
       "7                    Science           500    2014      Jaitin DA  \n",
       "8             Nat Biotechnol           483    2012     Ramsköld D  \n",
       "9                     Nature           477    2013      Shalek AK  \n",
       "10                    Nature           473    2014    Treutlein B  \n",
       "11                   Science           465    2016       Tirosh I  \n",
       "12       Nat Struct Mol Biol           451    2013          Yan L  \n",
       "13              Nat Neurosci           417    2015      Usoskin D  \n",
       "14            Nat Biotechnol           399    2015       Satija R  \n",
       "15               Genome Biol           387    2016       Conesa A  \n",
       "16                      Cell           378    2014           Wu H  \n",
       "17                  Cell Rep           365    2012  Hashimshony T  \n",
       "18                    Nature           346    2014      Shalek AK  \n",
       "19                Nat Commun           340    2017       Zheng GX  \n",
       "20                   Science           335    2014         Deng Q  \n",
       "21               Nat Methods           327    2014        Islam S  \n",
       "22                      Cell           322    2017  Keren-Shaul H  \n",
       "23                   Science           322    2013     Hackett JA  \n",
       "24                    Nature           321    2013          Xue Z  \n",
       "25                Genome Res           321    2011        Islam S  \n",
       "26              Nat Neurosci           308    2016        Tasic B  \n",
       "27                    Nature           305    2015  Buenrostro JD  \n",
       "28                      Cell           300    2014    Takashima Y  \n",
       "29               Nat Methods           294    2013    Brennecke P  \n",
       "..                       ...           ...     ...            ...  \n",
       "34             Nat Rev Genet           273    2015       Stegle O  \n",
       "35               Nat Methods           269    2014  Kharchenko PV  \n",
       "36            Cell Stem Cell           266    2010         Tang F  \n",
       "37                   Science           251    2016       Sharma U  \n",
       "38                Nat Protoc           233    2010         Tang F  \n",
       "39                   Science           231    2017     Villani AC  \n",
       "40            Cell Stem Cell           217    2015         Shin J  \n",
       "41            Cell Stem Cell           212    2014       Hogan BL  \n",
       "42                   Science           201    2015  Cusanovich DA  \n",
       "43         Nucleic Acids Res           199    2016    Petryszak R  \n",
       "44             Nat Rev Genet           197    2016      Scotti MM  \n",
       "45            Nat Biotechnol           197    2015        Rotem A  \n",
       "46            Cell Stem Cell           189    2016  Nowakowski TJ  \n",
       "47         Nucleic Acids Res           185    2014      Saliba AE  \n",
       "48                Genome Res           183    2014     Marinov GK  \n",
       "49                Cell Metab           177    2016  Segerstolpe Å  \n",
       "50                   Nat Med           176    2015    Alizadeh AA  \n",
       "51         Nucleic Acids Res           175    2018        Afgan E  \n",
       "52                      Cell           170    2016      Shekhar K  \n",
       "53               Nat Immunol           170    2016     Goldmann T  \n",
       "54                      Cell           163    2015      Pollen AA  \n",
       "55  Proc Natl Acad Sci U S A           161    2015        Camp JG  \n",
       "56               Genome Biol           160    2015        Finak G  \n",
       "57                    Nature           160    2015     Vaughan AE  \n",
       "58               Nat Methods           158    2017     Kiselev VY  \n",
       "59                      Cell           158    2016        Dixit A  \n",
       "60                  Mol Cell           157    2017   Ziegenhain C  \n",
       "61                    Nature           155    2014       Kumar RM  \n",
       "62            Cell Stem Cell           151    2015        Bond AM  \n",
       "63                PLoS Genet           151    2014    Macaulay IC  \n",
       "\n",
       "[64 rows x 7 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_first_author(string):\n",
    "    result = string.split(',')[0]\n",
    "    return(result)\n",
    "    \n",
    "first_authors = single_cell_papers['authorString'].apply(get_first_author)\n",
    "single_cell_papers['firstAuthor'] = first_authors\n",
    "single_cell_papers = single_cell_papers[single_cell_papers['citedByCount']>=150]\n",
    "\n",
    "single_cell_papers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cool, now we have our database of interest: all articles with more than 150 citations in EuropePMC about \"single-cell RNA sequencing\". \n",
    "\n",
    "The goal now is to turn this database into a markdown, user-friendly list. \n",
    "\n",
    "We will start by creating a base markdown file\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\n# Overview\\n\\n\\nLet's start lookinf for classic citations at the EuropePMC database.Once more, we will use naive definitions of citation classics.  \\n*Classics are all cited more than 150 times.  \\n*Big classics: more than 500 citations  \\n*Medium classics: at least 300 citations, less than 500  \\n*Small classics: at least 150 citations, less than 300\\n\""
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from mdutils.mdutils import MdUtils\n",
    "mdFile = MdUtils(file_name='README', title='Classic citations from the Europe PMC database')\n",
    "\n",
    "mdFile.new_header(level=1, title='Overview')\n",
    "mdFile.new_paragraph(\"Let's start lookinf for classic citations at the EuropePMC database.Once more, we will use naive definitions of citation classics.\") \n",
    "\n",
    "mdFile.new_line()\n",
    "mdFile.new_line(\"* Classics are all cited more than 150 times.\")\n",
    "mdFile.new_line(\"* Big classics: more than 500 citations\")\n",
    "mdFile.new_line(\"* Medium classics: at least 300 citations, less than 500\")\n",
    "mdFile.new_line(\"* Small classics: at least 150 citations, less than 300\\n\")\n",
    "\n",
    "\n",
    "mdFile.new_line(\"Currently, we have \" + str(single_cell_papers.shape[0]) + \" classics that meet the criteria above:\")\n",
    "mdFile.new_line()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_years(pd):\n",
    "    years = list(set(pd['pubYear']))\n",
    "    years.sort(reverse = True)\n",
    "    return(years)\n",
    "    \n",
    "    \n",
    "def get_new_entry(row):\n",
    "    result = (\"* \" + row['firstAuthor'] + \" et al, \" + \"[\" + row['title'] + \"]\" + \n",
    "         \"(\" + \"https://europepmc.org/article/MED/\" + row['pmid'] + \")\")\n",
    "    return(result)\n",
    "    \n",
    "    \n",
    "def add_section_to_md(pd, mdFile):\n",
    "    pd = pd.sort_values('firstAuthor')\n",
    "    years = get_years(pd)\n",
    "\n",
    "    for year in years:\n",
    "        mdFile.new_header(3, year)\n",
    "        df = pd[pd[\"pubYear\"] == year]\n",
    "        for index,row in df.iterrows():\n",
    "           mdFile.new_line(get_new_entry(row))\n",
    "        \n",
    "        mdFile.new_line()\n",
    "    \n",
    "    return(mdFile)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\n# Overview\\n\\n\\nLet's start lookinf for classic citations at the EuropePMC database.Once more, we will use naive definitions of citation classics.  \\n*Classics are all cited more than 150 times.  \\n*Big classics: more than 500 citations  \\n*Medium classics: at least 300 citations, less than 500  \\n*Small classics: at least 150 citations, less than 300\\n\\n# Big classics\\n\\n### 2015\\n  \\n* Klein AM et al, [Droplet barcoding for single-cell transcriptomics applied to embryonic stem cells.](https://europepmc.org/article/MED/26000487)  \\n* Macosko EZ et al, [Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets.](https://europepmc.org/article/MED/26000488)  \\n* Zeisel A et al, [Brain structure. Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq.](https://europepmc.org/article/MED/25700174)  \\n\\n### 2014\\n  \\n* Jaitin DA et al, [Massively parallel single-cell RNA-seq for marker-free decomposition of tissues into cell types.](https://europepmc.org/article/MED/24531970)  \\n* Lavin Y et al, [Tissue-resident macrophage enhancer landscapes are shaped by the local microenvironment.](https://europepmc.org/article/MED/25480296)  \\n* Patel AP et al, [Single-cell RNA-seq highlights intratumoral heterogeneity in primary glioblastoma.](https://europepmc.org/article/MED/24925914)  \\n* Picelli S et al, [Full-length RNA-seq from single cells using Smart-seq2.](https://europepmc.org/article/MED/24385147)  \\n* Trapnell C et al, [The dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells.](https://europepmc.org/article/MED/24658644)  \\n\\n# Medium classics\\n\\n### 2017\\n  \\n* Keren-Shaul H et al, [A Unique Microglia Type Associated with Restricting Development of Alzheimer's Disease.](https://europepmc.org/article/MED/28602351)  \\n* Zheng GX et al, [Massively parallel digital transcriptional profiling of single cells.](https://europepmc.org/article/MED/28091601)  \\n\\n### 2016\\n  \\n* Conesa A et al, [A survey of best practices for RNA-seq data analysis.](https://europepmc.org/article/MED/26813401)  \\n* Tasic B et al, [Adult mouse cortical cell taxonomy revealed by single cell transcriptomics.](https://europepmc.org/article/MED/26727548)  \\n* Tirosh I et al, [Dissecting the multicellular ecosystem of metastatic melanoma by single-cell RNA-seq.](https://europepmc.org/article/MED/27124452)  \\n\\n### 2015\\n  \\n* Buenrostro JD et al, [Single-cell chromatin accessibility reveals principles of regulatory variation.](https://europepmc.org/article/MED/26083756)  \\n* Satija R et al, [Spatial reconstruction of single-cell gene expression data.](https://europepmc.org/article/MED/25867923)  \\n* Usoskin D et al, [Unbiased classification of sensory neuron types by large-scale single-cell RNA sequencing.](https://europepmc.org/article/MED/25420068)  \\n\\n### 2014\\n  \\n* Deng Q et al, [Single-cell RNA-seq reveals dynamic, random monoallelic gene expression in mammalian cells.](https://europepmc.org/article/MED/24408435)  \\n* Islam S et al, [Quantitative single-cell RNA-seq with unique molecular identifiers.](https://europepmc.org/article/MED/24363023)  \\n* Shalek AK et al, [Single-cell RNA-seq reveals dynamic paracrine control of cellular variation.](https://europepmc.org/article/MED/24919153)  \\n* Takashima Y et al, [Resetting transcription factor control circuitry toward ground-state pluripotency in human.](https://europepmc.org/article/MED/25215486)  \\n* Treutlein B et al, [Reconstructing lineage hierarchies of the distal lung epithelium using single-cell RNA-seq.](https://europepmc.org/article/MED/24739965)  \\n* Wu H et al, [Reversing DNA methylation: mechanisms, genomics, and biological functions.](https://europepmc.org/article/MED/24439369)  \\n\\n### 2013\\n  \\n* Hackett JA et al, [Germline DNA demethylation dynamics and imprint erasure through 5-hydroxymethylcytosine.](https://europepmc.org/article/MED/23223451)  \\n* Shalek AK et al, [Single-cell transcriptomics reveals bimodality in expression and splicing in immune cells.](https://europepmc.org/article/MED/23685454)  \\n* Xue Z et al, [Genetic programs in human and mouse early embryos revealed by single-cell RNA sequencing.](https://europepmc.org/article/MED/23892778)  \\n* Yan L et al, [Single-cell RNA-Seq profiling of human preimplantation embryos and embryonic stem cells.](https://europepmc.org/article/MED/23934149)  \\n\\n### 2012\\n  \\n* Hashimshony T et al, [CEL-Seq: single-cell RNA-Seq by multiplexed linear amplification.](https://europepmc.org/article/MED/22939981)  \\n* Ramsköld D et al, [Full-length mRNA-Seq from single-cell levels of RNA and individual circulating tumor cells.](https://europepmc.org/article/MED/22820318)  \\n\\n### 2011\\n  \\n* Islam S et al, [Characterization of the single-cell transcriptional landscape by highly multiplex RNA-seq.](https://europepmc.org/article/MED/21543516)  \\n\\n# Small classics\\n\\n### 2018\\n  \\n* Afgan E et al, [The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update.](https://europepmc.org/article/MED/29790989)  \\n* Butler A et al, [Integrating single-cell transcriptomic data across different conditions, technologies, and species.](https://europepmc.org/article/MED/29608179)  \\n\\n### 2017\\n  \\n* Kiselev VY et al, [SC3: consensus clustering of single-cell RNA-seq data.](https://europepmc.org/article/MED/28346451)  \\n* Villani AC et al, [Single-cell RNA-seq reveals new types of human blood dendritic cells, monocytes, and progenitors.](https://europepmc.org/article/MED/28428369)  \\n* Ziegenhain C et al, [Comparative Analysis of Single-Cell RNA Sequencing Methods.](https://europepmc.org/article/MED/28212749)  \\n\\n### 2016\\n  \\n* Dixit A et al, [Perturb-Seq: Dissecting Molecular Circuits with Scalable Single-Cell RNA Profiling of Pooled Genetic Screens.](https://europepmc.org/article/MED/27984732)  \\n* Goldmann T et al, [Origin, fate and dynamics of macrophages at central nervous system interfaces.](https://europepmc.org/article/MED/27135602)  \\n* Nowakowski TJ et al, [Expression Analysis Highlights AXL as a Candidate Zika Virus Entry Receptor in Neural Stem Cells.](https://europepmc.org/article/MED/27038591)  \\n* Petryszak R et al, [Expression Atlas update--an integrated database of gene and protein expression in humans, animals and plants.](https://europepmc.org/article/MED/26481351)  \\n* Scotti MM et al, [RNA mis-splicing in disease.](https://europepmc.org/article/MED/26593421)  \\n* Segerstolpe Å et al, [Single-Cell Transcriptome Profiling of Human Pancreatic Islets in Health and Type 2 Diabetes.](https://europepmc.org/article/MED/27667667)  \\n* Sharma U et al, [Biogenesis and function of tRNA fragments during sperm maturation and fertilization in mammals.](https://europepmc.org/article/MED/26721685)  \\n* Shekhar K et al, [Comprehensive Classification of Retinal Bipolar Neurons by Single-Cell Transcriptomics.](https://europepmc.org/article/MED/27565351)  \\n\\n### 2015\\n  \\n* Alizadeh AA et al, [Toward understanding and exploiting tumor heterogeneity.](https://europepmc.org/article/MED/26248267)  \\n* Bond AM et al, [Adult Mammalian Neural Stem Cells and Neurogenesis: Five Decades Later.](https://europepmc.org/article/MED/26431181)  \\n* Camp JG et al, [Human cerebral organoids recapitulate gene expression programs of fetal neocortex development.](https://europepmc.org/article/MED/26644564)  \\n* Cusanovich DA et al, [Multiplex single cell profiling of chromatin accessibility by combinatorial cellular indexing.](https://europepmc.org/article/MED/25953818)  \\n* Finak G et al, [MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data.](https://europepmc.org/article/MED/26653891)  \\n* Pollen AA et al, [Molecular identity of human outer radial glia during cortical development.](https://europepmc.org/article/MED/26406371)  \\n* Rotem A et al, [Single-cell ChIP-seq reveals cell subpopulations defined by chromatin state.](https://europepmc.org/article/MED/26458175)  \\n* Shin J et al, [Single-Cell RNA-Seq with Waterfall Reveals Molecular Cascades underlying Adult Neurogenesis.](https://europepmc.org/article/MED/26299571)  \\n* Stegle O et al, [Computational and analytical challenges in single-cell transcriptomics.](https://europepmc.org/article/MED/25628217)  \\n* Vaughan AE et al, [Lineage-negative progenitors mobilize to regenerate lung epithelium after major injury.](https://europepmc.org/article/MED/25533958)  \\n\\n### 2014\\n  \\n* Bendall SC et al, [Single-cell trajectory detection uncovers progression and regulatory coordination in human B cell development.](https://europepmc.org/article/MED/24766814)  \\n* Hogan BL et al, [Repair and regeneration of the respiratory system: complexity, plasticity, and mechanisms of lung stem cell function.](https://europepmc.org/article/MED/25105578)  \\n* Kharchenko PV et al, [Bayesian approach to single-cell differential expression analysis.](https://europepmc.org/article/MED/24836921)  \\n* Kumar RM et al, [Deconstructing transcriptional heterogeneity in pluripotent stem cells.](https://europepmc.org/article/MED/25471879)  \\n* Macaulay IC et al, [Single cell genomics: advances and future perspectives.](https://europepmc.org/article/MED/24497842)  \\n* Marinov GK et al, [From single-cell to cell-pool transcriptomes: stochasticity in gene expression and RNA splicing.](https://europepmc.org/article/MED/24299736)  \\n* Saliba AE et al, [Single-cell RNA-seq: advances and future challenges.](https://europepmc.org/article/MED/25053837)  \\n* Theunissen TW et al, [Systematic identification of culture conditions for induction and maintenance of naive human pluripotency.](https://europepmc.org/article/MED/25090446)  \\n* Wu AR et al, [Quantitative assessment of single-cell RNA-sequencing methods.](https://europepmc.org/article/MED/24141493)  \\n\\n### 2013\\n  \\n* Brennecke P et al, [Accounting for technical noise in single-cell RNA-seq experiments.](https://europepmc.org/article/MED/24056876)  \\n\\n### 2010\\n  \\n* Tang F et al, [RNA-Seq analysis to capture the transcriptome landscape of a single cell.](https://europepmc.org/article/MED/20203668)  \\n* Tang F et al, [Tracing the derivation of embryonic stem cells from the inner cell mass by single-cell RNA-Seq analysis.](https://europepmc.org/article/MED/20452321)  \\n\\n# Contributions\\n  \\nWant to contribute a classic that I've missed? Great! Just add classic to a fork, make a pull request, and it is good to go.\""
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "mdFile.new_header(1, \"Big classics\")\n",
    "big_classics = single_cell_papers[single_cell_papers['citedByCount'] >= 500]\n",
    "mdFile = add_section_to_md(big_classics, mdFile)\n",
    "\n",
    "mdFile.new_header(1, \"Medium classics\")\n",
    "mid_classics = single_cell_papers[(single_cell_papers['citedByCount'] >= 300)\n",
    "                                  & (single_cell_papers['citedByCount'] < 500)]\n",
    "mdFile = add_section_to_md(mid_classics, mdFile)\n",
    "\n",
    "\n",
    "mdFile.new_header(1, \"Small classics\")\n",
    "smol_classics = single_cell_papers[(single_cell_papers['citedByCount'] >= 150)\n",
    "                                  & (single_cell_papers['citedByCount'] < 300)]\n",
    "mdFile = add_section_to_md(smol_classics, mdFile)\n",
    "\n",
    "mdFile.new_header(1, \"Contributions\")\n",
    "\n",
    "mdFile.new_line(\"Want to contribute a classic that I've missed? Great! Just add classic to a fork, make a pull request, and it is good to go.\")\n",
    "\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\n# Overview\\n\\n\\nLet's start lookinf for classic citations at the EuropePMC database.Once more, we will use naive definitions of citation classics.  \\n*Classics are all cited more than 150 times.  \\n*Big classics: more than 500 citations  \\n*Medium classics: at least 300 citations, less than 500  \\n*Small classics: at least 150 citations, less than 300\\n\\n# Big classics\\n\\n### 2015\\n  \\n* Klein AM et al, [Droplet barcoding for single-cell transcriptomics applied to embryonic stem cells.](https://europepmc.org/article/MED/26000487)  \\n* Macosko EZ et al, [Highly Parallel Genome-wide Expression Profiling of Individual Cells Using Nanoliter Droplets.](https://europepmc.org/article/MED/26000488)  \\n* Zeisel A et al, [Brain structure. Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq.](https://europepmc.org/article/MED/25700174)  \\n\\n### 2014\\n  \\n* Jaitin DA et al, [Massively parallel single-cell RNA-seq for marker-free decomposition of tissues into cell types.](https://europepmc.org/article/MED/24531970)  \\n* Lavin Y et al, [Tissue-resident macrophage enhancer landscapes are shaped by the local microenvironment.](https://europepmc.org/article/MED/25480296)  \\n* Patel AP et al, [Single-cell RNA-seq highlights intratumoral heterogeneity in primary glioblastoma.](https://europepmc.org/article/MED/24925914)  \\n* Picelli S et al, [Full-length RNA-seq from single cells using Smart-seq2.](https://europepmc.org/article/MED/24385147)  \\n* Trapnell C et al, [The dynamics and regulators of cell fate decisions are revealed by pseudotemporal ordering of single cells.](https://europepmc.org/article/MED/24658644)  \\n\\n# Medium classics\\n\\n### 2017\\n  \\n* Keren-Shaul H et al, [A Unique Microglia Type Associated with Restricting Development of Alzheimer's Disease.](https://europepmc.org/article/MED/28602351)  \\n* Zheng GX et al, [Massively parallel digital transcriptional profiling of single cells.](https://europepmc.org/article/MED/28091601)  \\n\\n### 2016\\n  \\n* Conesa A et al, [A survey of best practices for RNA-seq data analysis.](https://europepmc.org/article/MED/26813401)  \\n* Tasic B et al, [Adult mouse cortical cell taxonomy revealed by single cell transcriptomics.](https://europepmc.org/article/MED/26727548)  \\n* Tirosh I et al, [Dissecting the multicellular ecosystem of metastatic melanoma by single-cell RNA-seq.](https://europepmc.org/article/MED/27124452)  \\n\\n### 2015\\n  \\n* Buenrostro JD et al, [Single-cell chromatin accessibility reveals principles of regulatory variation.](https://europepmc.org/article/MED/26083756)  \\n* Satija R et al, [Spatial reconstruction of single-cell gene expression data.](https://europepmc.org/article/MED/25867923)  \\n* Usoskin D et al, [Unbiased classification of sensory neuron types by large-scale single-cell RNA sequencing.](https://europepmc.org/article/MED/25420068)  \\n\\n### 2014\\n  \\n* Deng Q et al, [Single-cell RNA-seq reveals dynamic, random monoallelic gene expression in mammalian cells.](https://europepmc.org/article/MED/24408435)  \\n* Islam S et al, [Quantitative single-cell RNA-seq with unique molecular identifiers.](https://europepmc.org/article/MED/24363023)  \\n* Shalek AK et al, [Single-cell RNA-seq reveals dynamic paracrine control of cellular variation.](https://europepmc.org/article/MED/24919153)  \\n* Takashima Y et al, [Resetting transcription factor control circuitry toward ground-state pluripotency in human.](https://europepmc.org/article/MED/25215486)  \\n* Treutlein B et al, [Reconstructing lineage hierarchies of the distal lung epithelium using single-cell RNA-seq.](https://europepmc.org/article/MED/24739965)  \\n* Wu H et al, [Reversing DNA methylation: mechanisms, genomics, and biological functions.](https://europepmc.org/article/MED/24439369)  \\n\\n### 2013\\n  \\n* Hackett JA et al, [Germline DNA demethylation dynamics and imprint erasure through 5-hydroxymethylcytosine.](https://europepmc.org/article/MED/23223451)  \\n* Shalek AK et al, [Single-cell transcriptomics reveals bimodality in expression and splicing in immune cells.](https://europepmc.org/article/MED/23685454)  \\n* Xue Z et al, [Genetic programs in human and mouse early embryos revealed by single-cell RNA sequencing.](https://europepmc.org/article/MED/23892778)  \\n* Yan L et al, [Single-cell RNA-Seq profiling of human preimplantation embryos and embryonic stem cells.](https://europepmc.org/article/MED/23934149)  \\n\\n### 2012\\n  \\n* Hashimshony T et al, [CEL-Seq: single-cell RNA-Seq by multiplexed linear amplification.](https://europepmc.org/article/MED/22939981)  \\n* Ramsköld D et al, [Full-length mRNA-Seq from single-cell levels of RNA and individual circulating tumor cells.](https://europepmc.org/article/MED/22820318)  \\n\\n### 2011\\n  \\n* Islam S et al, [Characterization of the single-cell transcriptional landscape by highly multiplex RNA-seq.](https://europepmc.org/article/MED/21543516)  \\n\\n# Small classics\\n\\n### 2018\\n  \\n* Afgan E et al, [The Galaxy platform for accessible, reproducible and collaborative biomedical analyses: 2018 update.](https://europepmc.org/article/MED/29790989)  \\n* Butler A et al, [Integrating single-cell transcriptomic data across different conditions, technologies, and species.](https://europepmc.org/article/MED/29608179)  \\n\\n### 2017\\n  \\n* Kiselev VY et al, [SC3: consensus clustering of single-cell RNA-seq data.](https://europepmc.org/article/MED/28346451)  \\n* Villani AC et al, [Single-cell RNA-seq reveals new types of human blood dendritic cells, monocytes, and progenitors.](https://europepmc.org/article/MED/28428369)  \\n* Ziegenhain C et al, [Comparative Analysis of Single-Cell RNA Sequencing Methods.](https://europepmc.org/article/MED/28212749)  \\n\\n### 2016\\n  \\n* Dixit A et al, [Perturb-Seq: Dissecting Molecular Circuits with Scalable Single-Cell RNA Profiling of Pooled Genetic Screens.](https://europepmc.org/article/MED/27984732)  \\n* Goldmann T et al, [Origin, fate and dynamics of macrophages at central nervous system interfaces.](https://europepmc.org/article/MED/27135602)  \\n* Nowakowski TJ et al, [Expression Analysis Highlights AXL as a Candidate Zika Virus Entry Receptor in Neural Stem Cells.](https://europepmc.org/article/MED/27038591)  \\n* Petryszak R et al, [Expression Atlas update--an integrated database of gene and protein expression in humans, animals and plants.](https://europepmc.org/article/MED/26481351)  \\n* Scotti MM et al, [RNA mis-splicing in disease.](https://europepmc.org/article/MED/26593421)  \\n* Segerstolpe Å et al, [Single-Cell Transcriptome Profiling of Human Pancreatic Islets in Health and Type 2 Diabetes.](https://europepmc.org/article/MED/27667667)  \\n* Sharma U et al, [Biogenesis and function of tRNA fragments during sperm maturation and fertilization in mammals.](https://europepmc.org/article/MED/26721685)  \\n* Shekhar K et al, [Comprehensive Classification of Retinal Bipolar Neurons by Single-Cell Transcriptomics.](https://europepmc.org/article/MED/27565351)  \\n\\n### 2015\\n  \\n* Alizadeh AA et al, [Toward understanding and exploiting tumor heterogeneity.](https://europepmc.org/article/MED/26248267)  \\n* Bond AM et al, [Adult Mammalian Neural Stem Cells and Neurogenesis: Five Decades Later.](https://europepmc.org/article/MED/26431181)  \\n* Camp JG et al, [Human cerebral organoids recapitulate gene expression programs of fetal neocortex development.](https://europepmc.org/article/MED/26644564)  \\n* Cusanovich DA et al, [Multiplex single cell profiling of chromatin accessibility by combinatorial cellular indexing.](https://europepmc.org/article/MED/25953818)  \\n* Finak G et al, [MAST: a flexible statistical framework for assessing transcriptional changes and characterizing heterogeneity in single-cell RNA sequencing data.](https://europepmc.org/article/MED/26653891)  \\n* Pollen AA et al, [Molecular identity of human outer radial glia during cortical development.](https://europepmc.org/article/MED/26406371)  \\n* Rotem A et al, [Single-cell ChIP-seq reveals cell subpopulations defined by chromatin state.](https://europepmc.org/article/MED/26458175)  \\n* Shin J et al, [Single-Cell RNA-Seq with Waterfall Reveals Molecular Cascades underlying Adult Neurogenesis.](https://europepmc.org/article/MED/26299571)  \\n* Stegle O et al, [Computational and analytical challenges in single-cell transcriptomics.](https://europepmc.org/article/MED/25628217)  \\n* Vaughan AE et al, [Lineage-negative progenitors mobilize to regenerate lung epithelium after major injury.](https://europepmc.org/article/MED/25533958)  \\n\\n### 2014\\n  \\n* Bendall SC et al, [Single-cell trajectory detection uncovers progression and regulatory coordination in human B cell development.](https://europepmc.org/article/MED/24766814)  \\n* Hogan BL et al, [Repair and regeneration of the respiratory system: complexity, plasticity, and mechanisms of lung stem cell function.](https://europepmc.org/article/MED/25105578)  \\n* Kharchenko PV et al, [Bayesian approach to single-cell differential expression analysis.](https://europepmc.org/article/MED/24836921)  \\n* Kumar RM et al, [Deconstructing transcriptional heterogeneity in pluripotent stem cells.](https://europepmc.org/article/MED/25471879)  \\n* Macaulay IC et al, [Single cell genomics: advances and future perspectives.](https://europepmc.org/article/MED/24497842)  \\n* Marinov GK et al, [From single-cell to cell-pool transcriptomes: stochasticity in gene expression and RNA splicing.](https://europepmc.org/article/MED/24299736)  \\n* Saliba AE et al, [Single-cell RNA-seq: advances and future challenges.](https://europepmc.org/article/MED/25053837)  \\n* Theunissen TW et al, [Systematic identification of culture conditions for induction and maintenance of naive human pluripotency.](https://europepmc.org/article/MED/25090446)  \\n* Wu AR et al, [Quantitative assessment of single-cell RNA-sequencing methods.](https://europepmc.org/article/MED/24141493)  \\n\\n### 2013\\n  \\n* Brennecke P et al, [Accounting for technical noise in single-cell RNA-seq experiments.](https://europepmc.org/article/MED/24056876)  \\n\\n### 2010\\n  \\n* Tang F et al, [RNA-Seq analysis to capture the transcriptome landscape of a single cell.](https://europepmc.org/article/MED/20203668)  \\n* Tang F et al, [Tracing the derivation of embryonic stem cells from the inner cell mass by single-cell RNA-Seq analysis.](https://europepmc.org/article/MED/20452321)  \\n\\n# Contributions\\n  \\nWant to contribute a classic that I've missed? Great! Just add classic to a fork, make a pull request, and it is good to go.\\n# Last update:\\n  \\nDecember 19, 2019\""
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from datetime import date\n",
    "today = date.today()\n",
    "today_long_format = today.strftime(\"%B %d, %Y\")\n",
    "mdFile.new_line()\n",
    "mdFile.new_header(1, \"Last update:\")\n",
    "\n",
    "mdFile.new_line(today_long_format)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<mdutils.fileutils.fileutils.MarkDownFile at 0x7fa02f644320>"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mdFile.create_md_file()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "64"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
