{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's start lookinf for classic citations at the EuropePMC database.\n",
    "\n",
    "Once more, we will use naive definitions of citation classics. \n",
    "\n",
    "Classics are all cited more than 150 times. \n",
    "\n",
    "Big classics: more than 500 citations\n",
    "\n",
    "Medium classics: at least 300 citations, less than 500\n",
    "\n",
    "Small classics: at least 150 citations, less than 300\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "from pandas.io.json import json_normalize\n",
    "\n",
    "\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "pmc_query = \"https://www.ebi.ac.uk/europepmc/webservices/rest/search?query=%22single-cell%20RNA-seq%22&resultType=lite&cursorMark=*&pageSize=100&sort=CITED%20desc&format=json\"\n",
    "page = requests.get(pmc_query)\n",
    "\n",
    "data = json.loads(page.text)\n",
    "df = json_normalize(data['resultList']['result'])\n",
    "\n",
    "\n",
    "single_cell_papers = df[['pmid', 'title', 'authorString', 'journalTitle', 'citedByCount', 'pubYear']]\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pmid</th>\n",
       "      <th>title</th>\n",
       "      <th>authorString</th>\n",
       "      <th>journalTitle</th>\n",
       "      <th>citedByCount</th>\n",
       "      <th>pubYear</th>\n",
       "      <th>firstAuthor</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>26000488</td>\n",
       "      <td>Highly Parallel Genome-wide Expression Profili...</td>\n",
       "      <td>Macosko EZ, Basu A, Satija R, Nemesh J, Shekha...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>1092</td>\n",
       "      <td>2015</td>\n",
       "      <td>Macosko EZ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>24925914</td>\n",
       "      <td>Single-cell RNA-seq highlights intratumoral he...</td>\n",
       "      <td>Patel AP, Tirosh I, Trombetta JJ, Shalek AK, G...</td>\n",
       "      <td>Science</td>\n",
       "      <td>938</td>\n",
       "      <td>2014</td>\n",
       "      <td>Patel AP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>25700174</td>\n",
       "      <td>Brain structure. Cell types in the mouse corte...</td>\n",
       "      <td>Zeisel A, Muñoz-Manchado AB, Codeluppi S, Lönn...</td>\n",
       "      <td>Science</td>\n",
       "      <td>726</td>\n",
       "      <td>2015</td>\n",
       "      <td>Zeisel A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>26000487</td>\n",
       "      <td>Droplet barcoding for single-cell transcriptom...</td>\n",
       "      <td>Klein AM, Mazutis L, Akartuna I, Tallapragada ...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>582</td>\n",
       "      <td>2015</td>\n",
       "      <td>Klein AM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>24658644</td>\n",
       "      <td>The dynamics and regulators of cell fate decis...</td>\n",
       "      <td>Trapnell C, Cacchiarelli D, Grimsby J, Pokhare...</td>\n",
       "      <td>Nat Biotechnol</td>\n",
       "      <td>569</td>\n",
       "      <td>2014</td>\n",
       "      <td>Trapnell C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>27984732</td>\n",
       "      <td>Perturb-Seq: Dissecting Molecular Circuits wit...</td>\n",
       "      <td>Dixit A, Parnas O, Li B, Chen J, Fulco CP, Jer...</td>\n",
       "      <td>Cell</td>\n",
       "      <td>158</td>\n",
       "      <td>2016</td>\n",
       "      <td>Dixit A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60</td>\n",
       "      <td>28212749</td>\n",
       "      <td>Comparative Analysis of Single-Cell RNA Sequen...</td>\n",
       "      <td>Ziegenhain C, Vieth B, Parekh S, Reinius B, Gu...</td>\n",
       "      <td>Mol Cell</td>\n",
       "      <td>157</td>\n",
       "      <td>2017</td>\n",
       "      <td>Ziegenhain C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>26431181</td>\n",
       "      <td>Adult Mammalian Neural Stem Cells and Neurogen...</td>\n",
       "      <td>Bond AM, Ming GL, Song H.</td>\n",
       "      <td>Cell Stem Cell</td>\n",
       "      <td>151</td>\n",
       "      <td>2015</td>\n",
       "      <td>Bond AM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>25471879</td>\n",
       "      <td>Deconstructing transcriptional heterogeneity i...</td>\n",
       "      <td>Kumar RM, Cahan P, Shalek AK, Satija R, DaleyK...</td>\n",
       "      <td>Nature</td>\n",
       "      <td>151</td>\n",
       "      <td>2014</td>\n",
       "      <td>Kumar RM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>24497842</td>\n",
       "      <td>Single cell genomics: advances and future pers...</td>\n",
       "      <td>Macaulay IC, Voet T.</td>\n",
       "      <td>PLoS Genet</td>\n",
       "      <td>151</td>\n",
       "      <td>2014</td>\n",
       "      <td>Macaulay IC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pmid                                              title  \\\n",
       "0   26000488  Highly Parallel Genome-wide Expression Profili...   \n",
       "1   24925914  Single-cell RNA-seq highlights intratumoral he...   \n",
       "2   25700174  Brain structure. Cell types in the mouse corte...   \n",
       "3   26000487  Droplet barcoding for single-cell transcriptom...   \n",
       "4   24658644  The dynamics and regulators of cell fate decis...   \n",
       "..       ...                                                ...   \n",
       "59  27984732  Perturb-Seq: Dissecting Molecular Circuits wit...   \n",
       "60  28212749  Comparative Analysis of Single-Cell RNA Sequen...   \n",
       "61  26431181  Adult Mammalian Neural Stem Cells and Neurogen...   \n",
       "62  25471879  Deconstructing transcriptional heterogeneity i...   \n",
       "63  24497842  Single cell genomics: advances and future pers...   \n",
       "\n",
       "                                         authorString    journalTitle  \\\n",
       "0   Macosko EZ, Basu A, Satija R, Nemesh J, Shekha...            Cell   \n",
       "1   Patel AP, Tirosh I, Trombetta JJ, Shalek AK, G...         Science   \n",
       "2   Zeisel A, Muñoz-Manchado AB, Codeluppi S, Lönn...         Science   \n",
       "3   Klein AM, Mazutis L, Akartuna I, Tallapragada ...            Cell   \n",
       "4   Trapnell C, Cacchiarelli D, Grimsby J, Pokhare...  Nat Biotechnol   \n",
       "..                                                ...             ...   \n",
       "59  Dixit A, Parnas O, Li B, Chen J, Fulco CP, Jer...            Cell   \n",
       "60  Ziegenhain C, Vieth B, Parekh S, Reinius B, Gu...        Mol Cell   \n",
       "61                          Bond AM, Ming GL, Song H.  Cell Stem Cell   \n",
       "62  Kumar RM, Cahan P, Shalek AK, Satija R, DaleyK...          Nature   \n",
       "63                               Macaulay IC, Voet T.      PLoS Genet   \n",
       "\n",
       "    citedByCount pubYear   firstAuthor  \n",
       "0           1092    2015    Macosko EZ  \n",
       "1            938    2014      Patel AP  \n",
       "2            726    2015      Zeisel A  \n",
       "3            582    2015      Klein AM  \n",
       "4            569    2014    Trapnell C  \n",
       "..           ...     ...           ...  \n",
       "59           158    2016       Dixit A  \n",
       "60           157    2017  Ziegenhain C  \n",
       "61           151    2015       Bond AM  \n",
       "62           151    2014      Kumar RM  \n",
       "63           151    2014   Macaulay IC  \n",
       "\n",
       "[64 rows x 7 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_first_author(string):\n",
    "    result = string.split(',')[0]\n",
    "    return(result)\n",
    "    \n",
    "first_authors = single_cell_papers['authorString'].apply(get_first_author)\n",
    "single_cell_papers['firstAuthor'] = first_authors\n",
    "single_cell_papers = single_cell_papers[single_cell_papers['citedByCount']>=150]\n",
    "\n",
    "single_cell_papers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cool, now we have our database of interest: all articles with more than 150 citations in EuropePMC about \"single-cell RNA sequencing\". \n",
    "\n",
    "The goal now is to turn this database into a markdown, user-friendly list. \n",
    "\n",
    "We will start by creating a base markdown file\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\n# Overview\\n\\n\\nLet's start lookinf for classic citations at the EuropePMC database.Once more, we will use naive definitions of citation classics.  \\n*Classics are all cited more than 150 times.  \\n*Big classics: more than 500 citations  \\n*Medium classics: at least 300 citations, less than 500  \\n*Small classics: at least 150 citations, less than 300\\n\""
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from mdutils.mdutils import MdUtils\n",
    "mdFile = MdUtils(file_name='EuropePCM_classics', title='Classic citations from the Europe PMC database')\n",
    "\n",
    "mdFile.new_header(level=1, title='Overview')\n",
    "mdFile.new_paragraph(\"Let's start lookinf for classic citations at the EuropePMC database.Once more, we will use naive definitions of citation classics.\") \n",
    "\n",
    "\n",
    "mdFile.new_line(\"*Classics are all cited more than 150 times.\")\n",
    "mdFile.new_line(\"*Big classics: more than 500 citations\")\n",
    "mdFile.new_line(\"*Medium classics: at least 300 citations, less than 500\")\n",
    "mdFile.new_line(\"*Small classics: at least 150 citations, less than 300\\n\")\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_years(pd):\n",
    "    years = list(set(pd['pubYear']))\n",
    "    years.sort(reverse = True)\n",
    "    return(years)\n",
    "    \n",
    "    \n",
    "def get_new_entry(row):\n",
    "    result = (\"* \" + row['firstAuthor'] + \" et al, \" + \"[\" + row['title'] + \"]\" + \n",
    "         \"(\" + \"https://europepmc.org/article/MED/\" + row['pmid'] + \")\")\n",
    "    return(result)\n",
    "    \n",
    "    \n",
    "def add_section_to_md(pd, mdFile):\n",
    "    pd = pd.sort_values('firstAuthor')\n",
    "    years = get_years(pd)\n",
    "\n",
    "    for year in years:\n",
    "        mdFile.new_header(3, year)\n",
    "        df = pd[pd[\"pubYear\"] == year]\n",
    "        for index,row in df.iterrows():\n",
    "           mdFile.new_line(get_new_entry(row))\n",
    "        \n",
    "        mdFile.new_line()\n",
    "    \n",
    "    return(mdFile)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "mdFile.new_header(1, \"Big classics\")\n",
    "big_classics = single_cell_papers[single_cell_papers['citedByCount'] >= 500]\n",
    "mdFile = add_section_to_md(big_classics, mdFile)\n",
    "\n",
    "mdFile.new_header(1, \"Medium classics\")\n",
    "mid_classics = single_cell_papers[(single_cell_papers['citedByCount'] >= 300)\n",
    "                                  & (single_cell_papers['citedByCount'] < 500)]\n",
    "mdFile = add_section_to_md(mid_classics, mdFile)\n",
    "\n",
    "\n",
    "mdFile.new_header(1, \"Small classics\")\n",
    "smol_classics = single_cell_papers[(single_cell_papers['citedByCount'] >= 150)\n",
    "                                  & (single_cell_papers['citedByCount'] < 300)]\n",
    "mdFile = add_section_to_md(smol_classics, mdFile)\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<mdutils.fileutils.fileutils.MarkDownFile at 0x7fcf76214d30>"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mdFile.create_md_file()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
